src/main/java/seedu/address/model/LeavesBook.java

```diff
@@ -0,0 +1,125 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.leave.Leave;
+import seedu.address.model.leave.UniqueLeaveList;
+
+/**
+ * Wraps all data at the leaves-book level
+ * Duplicates are not allowed (by .isSameLeave comparison)
+ */
+public class LeavesBook implements ReadOnlyLeavesBook {
+
+    private final UniqueLeaveList leaves;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        leaves = new UniqueLeaveList();
+    }
+
+    public LeavesBook() {}
+
+    /**
+     * Creates an LeavesBook using the Leaves in the {@code toBeCopied}
+     */
+    public LeavesBook(ReadOnlyLeavesBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
+
+    /**
+     * Replaces the contents of the leave list with {@code leaves}.
+     * {@code leaves} must not contain duplicate leaves.
+     */
+    public void setLeaves(List<Leave> leaves) {
+        this.leaves.setLeaves(leaves);
+    }
+
+    /**
+     * Resets the existing data of this {@code LeavesBook} with {@code newData}.
+     */
+    public void resetData(ReadOnlyLeavesBook newData) {
```

Comment:
Does resetData clear all data or does it replace all data in the LeavesBook?

Reply:
resetData replaces all data in the leavesBook with newData